core/src/main/java/com/linecorp/armeria/server/annotation/BeanRequestConverterFunction.java

```diff
@@ -0,0 +1,320 @@
+
+            try {
+                constructor = beanClass.getDeclaredConstructor();
+            } catch (final NoSuchMethodException e) {
+                throw new IllegalArgumentException(
+                        "Cannot find default constructor for '" + beanClass.getCanonicalName() + "'.",
+                        e);
+            }
+
+            constructor.setAccessible(true);
+        }
+
+        private Object createInstance() {
+            try {
+                return constructor.newInstance();
+            } catch (final Exception e) {
+                throw new IllegalArgumentException(
+                        "Cannot create instance of '" + beanClass.getCanonicalName() + "'.",
+                        e);
+            }
+        }
+    }
+
+    private abstract static class ParamInfo {
+        private final Class<?> type;
+
+        private final ParamValueRetriever paramValueRetriever;
+
+        private final EnumConverter<?> enumConverter;
+
+        private ParamInfo(final Class<?> type,
+                          final ParamValueRetriever paramValueRetriever) {
+            this.type = validateAndNormalizeSupportedType(type);
+            this.paramValueRetriever = paramValueRetriever;
+
+            if (type.isEnum()) {
+                enumConverter = new EnumConverter<>(this.type.asSubclass(Enum.class));
+            } else {
+                enumConverter = null;
+            }
+        }
+
+        protected abstract void setParamValue(Object targetBean,
+                                              Object paramValue);
+
+        private void setValue(final ServiceRequestContext ctx,
+                              final AggregatedHttpMessage request,
+                              final MutableValueHolder<HttpParameters> httpParams,
+                              final Object targetBean) {
+            final String strValue = paramValueRetriever.getParamValue(ctx, request, httpParams);
+
+            if (strValue == null) {
+                return;
+            }
+
+            final Object paramValue = convertParameter(strValue, type, enumConverter, false);
+
+            setParamValue(targetBean, paramValue);
+        }
+    }
+
+    private abstract static class FieldParamInfo extends ParamInfo {
+        private final Field field;
+
+        private FieldParamInfo(final Field field,
+                               final ParamValueRetriever paramValueRetriever) {
+            super(field.getType(), paramValueRetriever);
+            this.field = field;
+            this.field.setAccessible(true);
+        }
+
+        @Override
+        protected void setParamValue(final Object targetBean,
+                                     final Object paramValue) {
+            try {
+                field.set(targetBean, paramValue);
+            } catch (final Exception e) {
+                throw new IllegalArgumentException(
+                        "Can't set '" + paramValue + "' " + "to field '" +
```

Comment:
`paramValue + "' " + "to field '"`  -> `paramValue + "' to field '"`

Reply:
these code was changed to `String.format('message template', msgParams)`